README.md

```diff
@@ -156,222 +145,144 @@ pluginsync=true
 
-```
-node "default" {
-    class { "datadog_agent":
-        api_key => "<YOUR_DD_API_KEY>",
+    ```conf
+    node "default" {
+        class { "datadog_agent":
+            api_key => "<YOUR_DD_API_KEY>",
+        }
     }
-}
-node "puppetmaster" {
-    class { "datadog_agent":
-        api_key            => "<YOUR_DD_API_KEY>",
-        puppet_run_reports => true
+    node "puppetmaster" {
+        class { "datadog_agent":
+            api_key            => "<YOUR_DD_API_KEY>",
+            puppet_run_reports => true
+        }
     }
-}
-```
-
-_Note: This may be file `/etc/puppet/manifests/nodes.pp` on older puppets_
-
-
-Run Puppet Agent
-----------------
+    ```
 
-```
-sudo systemctl restart puppetserver
-sudo puppet agent --onetime --no-daemonize --no-splay --verbose
-```
+     **Note**: For older versions of Puppet, edit `/etc/puppet/manifests/nodes.pp`.
 
-Example response:
+3. Run the Puppet Agent:
 
-```
-info: Retrieving plugin
-info: Caching catalog for alq-linux.dev.datadoghq.com
-info: Applying configuration version '1333470114'
-notice: Finished catalog run in 0.81 seconds
-```
+    ```shell
+    sudo systemctl restart puppetserver
+    sudo puppet agent --onetime --no-daemonize --no-splay --verbose
+    ```
 
-Verify on Datadog
------------------
+     Example response:
 
-1. Search for `puppet` on [the Integrations page][5]. The Puppet integration tile displays the install status.
+    ```text
+    info: Retrieving plugin
+    info: Caching catalog for alq-linux.dev.datadoghq.com
+    info: Applying configuration version '1333470114'
+    notice: Finished catalog run in 0.81 seconds
+    ```
 
-2. Search for `sources:puppet` in the [Event Stream][6] to see your Puppet events.
+4. Verify your Puppet data is in Datadog by searching for `sources:puppet` in the [Event Stream][5].
 
-Masterless Puppet
-=================
+## Masterless Puppet
 
-To use this specific setup, see https://gist.github.com/LeoCavaille/cd412c7a9ff5caec462f. This applies to older puppets and is untested on >=4.x Puppet versions.
+1. The Datadog module and its dependencies have to be installed on all nodes running masterless.
+2. Add this to each node's `site.pp` file:
+    ```conf
+    class { "datadog_agent":
+        api_key            => "<YOUR_DD_API_KEY>",
+        puppet_run_reports => true
+    }
+   ```
 
+3. Configure reports in the `[main]` section of `puppet.conf`:
+    ```conf
+    [main]
+    reports=datadog_reports
+    ```
+4. Run puppet in masterless configuration:
+    ```shell
+    puppet apply --modulepath <path_to_modules> <path_to_site.pp>
+    ```
 
-Client Settings
-===============
+## Client settings
 
 ### Tagging client nodes
 
 The Datadog Agent configuration file is recreated from the template every Puppet run. If you need to tag your nodes, add an array entry in Hiera:
 
-```
+```conf
 datadog_agent::tags:
 - 'keyname:value'
 - 'anotherkey:%{factname}'
 ```
 
+### Extra configuration variables
```

Comment:
```suggestion
### Configuration variables
```
'extra' makes they seem like not needed/additional. The most used config options and some required ones like `site` if using Datadog in EU are in this section, so we want people to read this.